src/studio/src/designer/frontend/shared/features/dataModelling/DataModelling.tsx

```diff
@@ -122,16 +122,23 @@ function DataModelling({
     dispatch(DataModelsMetadataActions.getDataModelsMetadata());
   };
 
+  const loadingIndicator = (
+    <AltinnSpinner
+      spinnerText={getLanguageFromKey('general.loading', language)}
+    />
+  );
   return (
     <SchemaEditor
       language={language}
       schema={jsonSchema}
       onSaveSchema={handleSaveSchema}
       name={selectedOption?.label}
       LoadingComponent={
-        <AltinnSpinner
-          spinnerText={getLanguageFromKey('general.loading', language)}
-        />
+        metadataLoadingState === LoadingState.LoadingModels ? (
```

Comment:
Could this check be part of the `loadingIndicator` definition instead?

Reply:
Dette skjønner jeg ikke helt. Forøvrig ville jeg nok heller lagd en attributt på `<SchemaEditor/>`-komponenten som en enkel `loading` true/false og latt komponenten tatt seg av å vise loading state. Men antar at det er ønsker om å kunne bytte loading indikatorer fra utsiden av komponenten her, så har ikke foreslått refaktorering.

Utrykket `metadataLoadingState === LoadingState.LoadingModels` ville jeg da bare latt ligget nærme komponenten for å gjøre det enklere å følge tråden i koden.

Reply:
Vil tro det ble implementert på denne måten for å unngå å knytte schema-editor koden sterkt til studio (shared).
Kanskje en grei løsning her er å sende med en ny prop `loading` i tillegg til loadingspinneren da, og la SchemaEditor komponenten styre visning.